wknml/__init__.py

```diff
@@ -220,90 +239,110 @@ def parse_comment(nml_comment):
 
 
 def parse_group(nml_group):
-    return Group(int(nml_group.get("id")), nml_group.get("name", default=""))
+    return Group(int(nml_group.get("id")), nml_group.get("name", default=""), [])
 
 
-def parse_nml(nml_root):
-
-    groups = [Group(id=1, name="")]
-    if nml_root.find("groups") is not None:
-      groups = [parse_group(g) for g in nml_root.find("groups")]
-
+def parse_nml(filename):
```

Comment:
@daniel-wer I think `parse_nml` shouldn't take a filename, but a file object which makes it more composable e.g. when using gzip or network-streams.